LoRaWAN Code won't linking when config with CN470 region

### prj.conf
CONFIG_LOG=y
CONFIG_SPI=y
CONFIG_GPIO=y
CONFIG_LORA=y
CONFIG_LORA_SX12XX=y
CONFIG_LORAWAN=y
CONFIG_LORAMAC_REGION_CN470=y
CONFIG_MAIN_STACK_SIZE=2048
CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
### output
`[165/175] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /Users/william/zephyrproject/zephyr/samples/subsys/lorawan/class_a/build/zephyr/zephyr_pre0.map 
: && ccache /Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc   zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/Users/william/zephyrproject/zephyr/samples/subsys/lorawan/class_a/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/newlib/liblib__libc__newlib.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/subsys/lorawan/libloramac-node.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/spi/libdrivers__spi.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  modules/stm32/stm32cube/lib..__modules__hal__stm32__stm32cube.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/thumb/v7e-m/nofp"  -L/Users/william/zephyrproject/zephyr/samples/subsys/lorawan/class_a/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m4  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -lm  -Wl,-lc  -L"/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/arm-zephyr-eabi"/lib/thumb/v7e-m/nofp  -Wl,-lgcc  -lc  -specs=nano.specs && cd /Users/william/zephyrproject/zephyr/samples/subsys/lorawan/class_a/build/zephyr && /usr/local/Cellar/cmake/3.24.1/bin/cmake -E echo
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/lorawan/libloramac-node.a(RegionCN470.c.obj): in function `RegionCN470InitDefaults':
/Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:576: undefined reference to `RegionCN470A20InitializeChannels'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20GetDownlinkFrequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20GetBeaconChannelOffset'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20LinkAdrChMaskUpdate'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20VerifyRfFreq'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20InitializeChannels'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20InitializeChannelsMask'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20GetRx1Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:597: undefined reference to `RegionCN470A20GetRx2Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/subsys/lorawan/libloramac-node.a(RegionCN470.c.obj): in function `RegionCN470ApplyCFList':
/Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20GetDownlinkFrequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20GetBeaconChannelOffset'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20LinkAdrChMaskUpdate'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20VerifyRfFreq'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20InitializeChannels'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20InitializeChannelsMask'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20GetRx1Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A20GetRx2Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20GetDownlinkFrequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20GetBeaconChannelOffset'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20LinkAdrChMaskUpdate'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20VerifyRfFreq'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20InitializeChannels'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20InitializeChannelsMask'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20GetRx1Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B20GetRx2Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26GetDownlinkFrequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26GetBeaconChannelOffset'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26LinkAdrChMaskUpdate'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26VerifyRfFreq'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26InitializeChannels'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26InitializeChannelsMask'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26GetRx1Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470A26GetRx2Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26GetDownlinkFrequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26GetBeaconChannelOffset'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26LinkAdrChMaskUpdate'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26VerifyRfFreq'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26InitializeChannels'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26InitializeChannelsMask'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26GetRx1Frequency'
/Users/william/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: /Users/william/zephyrproject/modules/lib/loramac-node/src/mac/region/RegionCN470.c:670: undefined reference to `RegionCN470B26GetRx2Frequency'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /Users/william/zephyrproject/zephyr/samples/subsys/lorawan/class_a/build`

--- a/modules/loramac-node/CMakeLists.txt
+++ b/modules/loramac-node/CMakeLists.txt
@@ -78,7 +78,12 @@ zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_AS923
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionAS923.c
 )
 zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_CN470
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionBaseUS.c
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionCN470.c
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionCN470A20.c
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionCN470A26.c
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionCN470B20.c
+  ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionCN470B26.c
 )
 zephyr_library_sources_ifdef(CONFIG_LORAMAC_REGION_KR920
   ${ZEPHYR_LORAMAC_NODE_MODULE_DIR}/src/mac/region/RegionKR920.c

